modules/azure/nsg.go

```diff
@@ -0,0 +1,194 @@
+	DestinationAddressPrefix string
+	Access                   string
+	Priority                 int32
+	Direction                string
+}
+
+func bindRuleList(dest *[]NsgRuleSummary, source network.SecurityRuleListResultIterator) {
+	for source.NotDone() {
+		v := source.Value()
+		*dest = append(*dest, convertToNsgRuleSummary(v.Name, v.SecurityRulePropertiesFormat))
+		err := source.NextWithContext(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func convertToNsgRuleSummary(name *string, rule *network.SecurityRulePropertiesFormat) NsgRuleSummary {
+	summary := NsgRuleSummary{}
+
+	if rule.Description != nil {
+		summary.Description = *rule.Description
+	} else {
+		summary.Description = ""
+	}
+
+	summary.Name = *name
+	summary.Protocol = string(rule.Protocol)
+	summary.SourcePortRange = *rule.SourcePortRange
+	summary.DestinationPortRange = *rule.DestinationPortRange
+	summary.SourceAddressPrefix = *rule.SourceAddressPrefix
+	summary.DestinationAddressPrefix = *rule.DestinationAddressPrefix
+	summary.Access = string(rule.Access)
+	summary.Priority = *rule.Priority
+	summary.Direction = string(rule.Direction)
+	return summary
+}
+
+// FindRuleByName looks for a matching rule name
+func (summarizedRules *NsgRuleSummaryList) FindRuleByName(name string) NsgRuleSummary {
+	for _, r := range summarizedRules.SummarizedRules {
+		if r.Name == name {
+			return r
+		}
+	}
+
+	return NsgRuleSummary{}
+}
+
+// AllowsDestinationPort checks to see if the rule allows a specific destination port.
+func (summarizedRule *NsgRuleSummary) AllowsDestinationPort(port string) bool {
+	if summarizedRule.DestinationPortRange == "*" {
+		return true
+	}
+
+	low, high := ParsePortRangeString(summarizedRule.DestinationPortRange)
+	portAsInt, _ := strconv.ParseInt(port, 10, 16)
+
+	if (port == "*") && (low == 0) && (high == 65535) {
+		return true
+	}
+
+	return ((low <= uint16(portAsInt)) && (uint16(portAsInt) <= high))
+}
+
+// AllowsSourcePort checks to see if the rule allows a specific source port.
+func (summarizedRule *NsgRuleSummary) AllowsSourcePort(port string) bool {
+	if summarizedRule.SourcePortRange == "*" {
+		return true
+	}
+
+	low, high := ParsePortRangeString(summarizedRule.SourcePortRange)
+	portAsInt, _ := strconv.ParseInt(port, 10, 16)
+
+	if (port == "*") && (low == 0) && (high == 65535) {
+		return true
+	}
+
+	return ((low <= uint16(portAsInt)) && (uint16(portAsInt) <= high))
+}
+
+// ParsePortRangeString decodes a range string ("2-100") or a single digit ("22") and returns
+// a tuple in [low, hi] form. Note that if a single digit is supplied, both members of the
+// return tuple will be the same value (e.g., "22" returns (22, 22))
+func ParsePortRangeString(rangeString string) (low uint16, hight uint16) {
+	// Is this an asterisk?
+	if rangeString == "*" {
+		return uint16(0), uint16(65535)
+	}
+
+	// Is this a range?
+	if !strings.Contains(rangeString, "-") {
+		val, _ := strconv.ParseInt(rangeString, 10, 16)
```

Comment:
This method ignores a lot of parsing errors. That could lead to a lot of swallowed errors that cause problems down the line that will be hard to debug. I'd recommend returning errors from this method instead to indicate the range string was invalid.

Reply:
Handled and propagated parse errors throughout this method.